Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts

```diff
@@ -9,196 +9,87 @@ import tl = require('vsts-task-lib/task');
+import JobQueue = jobqueue.JobQueue;
+import util = require('./util');
 
-function createLinkAndFinish(result, jobStatus: string, resultMessage: string) {
-    var tempDir = shell.tempdir();
-    var linkMarkdownFile = path.join(tempDir, 'JenkinsJob_' + jenkinsTaskName + '_' + jenkinsExecutableNumber + '.md');
-    tl.debug('jenkinsLink: ' + linkMarkdownFile);
-    var summaryTitle = 'Jenkins ' + jenkinsTaskName + ' - ' + jenkinsExecutableNumber + ' - ' + jobStatus;
-    tl.debug('summaryTitle: ' + summaryTitle);
-    var markdownContents = '[' + jenkinsExecutableUrl + '](' + jenkinsExecutableUrl + ')';
-    fs.writeFile(linkMarkdownFile, markdownContents, function callBack(err) {
-        if (err) {
-            //don't fail the build -- there just won't be a link
-            console.log('Error creating link to Jenkins job: ' + err);
-        } else {
-            console.log('##vso[task.addattachment type=Distributedtask.Core.Summary;name=' + summaryTitle + ';]' + linkMarkdownFile);
-        }
-        tl.setResult(result, resultMessage);
-    });
-}
+export class TaskOptions {
+    serverEndpoint = tl.getInput('serverEndpoint', true);
+    serverEndpointUrl = tl.getEndpointUrl(this.serverEndpoint, false);
 
-function captureJenkinsConsole(consoleOffset: number) {
-    var fullUrl = jenkinsExecutableUrl + '/logText/progressiveText/?start=' + consoleOffset;
-    tl.debug('Tracking progress of job URL: ' + fullUrl);
-    request.get({ url: fullUrl }, function callBack(err, httpResponse, body) {
-        if (err) {
-            tl.setResult(tl.TaskResult.Failed, err);
-        } else if (httpResponse.statusCode != 200) {
-            failReturnCode(httpResponse, 'Job progress tracking failed to read job progress');
-        } else {
-            console.log(body); // redirect Jenkins console to task console
-            var xMoreData = httpResponse.headers['x-more-data'];
-            if (xMoreData && xMoreData == 'true') {
-                var offset = httpResponse.headers['x-text-size'];
-                // job still running so keep logging console
-                setTimeout(function () {
-                    captureJenkinsConsole(offset);
-                }, captureConsolePollInterval);
-            } else { // job is done -- did it succeed or not?
-                checkSuccess();
-            }
-        }
-    });
-}
+    serverEndpointAuth = tl.getEndpointAuthorization(this.serverEndpoint, false);
+    username = this.serverEndpointAuth['parameters']['username'];
+    password = this.serverEndpointAuth['parameters']['password'];
+
+    jobName: string = tl.getInput('jobName', true);
+
+    captureConsole: boolean = tl.getBoolInput('captureConsole', true);
+    // capturePipeline is only possible if captureConsole mode is enabled
+    capturePipeline: boolean = this.captureConsole ? tl.getBoolInput('capturePipeline', true) : false;
+
+    pollInterval: number = 5000; // five seconds is what the Jenkins Web UI uses
```

Comment:
Rename this to something like `pollIntervalMilliseconds` so it's clear that milliseconds are the unit of measure.